tests/subsys/zigbee/osif/serial/serial_async_api/src/main.c

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <ztest.h>
+#include <zboss_api.h>
+
+#include <zb_nrf_serial.c>
+
+
+#define TEST_UART_BUF_LEN 256
+#define TEST_RX_LOCK_TIMEOUT K_MSEC(CONFIG_ZIGBEE_UART_RX_TIMEOUT + 100)
+#define TEST_TX_LOCK_TIMEOUT K_MSEC(100)
+
+
+K_SEM_DEFINE(tx_lock, 1, 1);
+K_SEM_DEFINE(rx_lock, 0, 1);
+
+static u8_t uart_rx_test_buf[TEST_UART_BUF_LEN];
+static u32_t uart_rx_buf_cnt;
+static u8_t uart_rx_mem_test_buf[TEST_UART_BUF_LEN];
+
+
+static size_t fill_with_test_data(u8_t *buf, size_t len)
+{
+	for (size_t i = 0; i < len; i++) {
+		buf[i] = (i % 0x100);
+	}
+
+	return len;
+}
+
+static void uart_recv_cb(zb_uint8_t *buf, zb_ushort_t len)
+{
+	if (uart_rx_buf_cnt + len <= sizeof(uart_rx_test_buf)) {
+		memcpy(&uart_rx_test_buf[uart_rx_buf_cnt], buf, len);
+		uart_rx_buf_cnt += len;
+	}
+	k_sem_give(&rx_lock);
+}
+
+static void uart_send_cb(zb_uint8_t status)
+{
+	k_sem_give(&tx_lock);
+}
+
+
```

Comment:
One empty line is enough.

Reply:
If this is not necessary, I'd prefer to keep an additional empty line to split visually static helper functions from test cases.